src/relay/backend/build_module.cc

```diff
@@ -473,23 +517,25 @@ class RelayBuildModule : public runtime::ModuleNode {
 
     // Relay IRModule -> IRModule optimizations.
     relay_module = Optimize(relay_module, targets_, params);
+
     // Get the updated function.
     auto func = Downcast<Function>(relay_module->Lookup("main"));
 
     // Generate code for the updated function.
-    graph_codegen_ = std::unique_ptr<GraphCodegen>(new GraphCodegen());
-    graph_codegen_->Init(nullptr, targets_);
-    graph_codegen_->Codegen(func);
-
-    ret_.graph_json = graph_codegen_->GetJSON();
-    ret_.params = graph_codegen_->GetParams();
+    const String executor_str =
+        target_host->GetAttr<String>("executor").value_or(kTvmExecutorGraph);
```

Comment:
I think "executor" should technically not be attribute of the target_host rather a parameter for relay.build(...). I think we should just plumb it through relay.build(.., executor=graph) while defaulting to graph -- thus its not a breaking change. 
cc : @areusch

Reply:
I agree. I think there are a set of options we've been putting into Target which don't belong. `--runtime` is another great example. I think we need to create a separate config object e.g. `deploy_env` which could be a parameter to `tvm.relay.build()`. i'm not sure we need to address this in this PR, since i think it will be a fairly invasive change.

Reply:
Alright! copying @mbaret . 
@giuseros did you have to workaround something because of this ?

Reply:
So, this change was quite straightforward to make. I just pushed it: see how it looks

Reply:
Ups, I didn't update the page, and didn't see the comments :) @areusch I was seeing a failure in a gpu test. The reason was the following:
* In order to understand  what `FactoryModule` I need to use, I query the `get_executor_type()` method of the `BuildModule` c++ object. 
* `get_executor_type`  is querying the `target_host`, and then does:
`executor_str = target_host->GetAttr<String>("executor").value_or(kTvmExecutorGraph)`
* In case of `target = "cuda"` the `target_host` is not defined and it was crashing. 
Two possible solutions:
* I can work around that by checking for `target_host.defined()` and default to "graph". 
* I can follow what Manupa suggests and move the `executor` as a build parameter
As I said, the change is not invasive, because the only place where I need the executor is within `build_module.cc`. Please, have a look and let me know if you prefer this or revert to a target option. 

My opinion is that, since it looks like not an invasive change, we can have it as a build parameter in this PR. What do you think?

Reply:
i agree this isn't too invasive now, the main question is whether `executor` should become a top-level `tvm.relay.build` parameter or whether e.g. `runtime_config` or a similarly named struct should be there instead. I feel like at least these parameters may not belong with target (and there are probably more, these just come to mind):
- executor
- runtime
- link-params (this one is arguable, but I don't think link-params fully describes the param placement)

I think it would be great to have another PR which addresses the question of: how do we model runtime config in TVM targets, and how should changes there affect autotuning schedule keys? The last bit is the contentious part. For this PR, perhaps we could take the approach I did with runtime: define it as a Target attribute with no default, and elsewhere make the default when unspecified kTvmExecutorGraph. This way, in the usual non-aot case, we don't affect tuning logs. wdyt?

Reply:
cc @jroesch @tqchen